src/emulator/apphosting/utils.ts

```diff
@@ -2,6 +2,7 @@ import { pathExists } from "fs-extra";
 import { join } from "path";
```

Comment:
High level comment - even though we do it all over the CLI codebase, `utils` files tend to be a antipattern - over time, they fill up with random stuff and that leads to a disorganized codebase. 

Right now, this file has 2 functions that detect what package manager is used. Consider renaming this to 'detectPackageManager.ts' or something to that effect?